releasetool/github.py

```diff
@@ -54,14 +67,71 @@ def _find_devrel_api_key() -> str:
     return magic_github_proxy_key
 
 
+class GitHubToken:
+    def __init__(self, token: Union[str, dict]):
+        self.auth_type = "Bearer"
+        # If a dictionary is provided for token, assume it
+        # contains app_id, installation, private_key, such that we
+        # can fetch a JWT:
+        if type(token) is dict:
```

Comment:
Personally, I would prefer to push this logic onto the caller -- in this case all the way up at the click command (where the option of using a single bearer token or app credentials is allowed). Otherwise we're propagating the Union[str, dict] type through a few calls

Reply:
I would defer to other python folks for this though.